src/CommandHandler/Wishlist/AddSelectedProductsToCartHandler.php

```diff
@@ -49,22 +50,25 @@ public function __construct(
     public function __invoke(AddSelectedProductsToCart $addSelectedProductsToCartCommand): void
     {
         $this->addSelectedProductsToCart($addSelectedProductsToCartCommand->getWishlistProducts());
-
-        $this->flashBag->add('success', $this->translator->trans('bitbag_sylius_wishlist_plugin.ui.added_selected_wishlist_items_to_cart'));
     }
 
     private function addSelectedProductsToCart(Collection $wishlistProducts): void
     {
         /** @var WishlistItem $wishlistProduct */
         foreach ($wishlistProducts as $wishlistProduct) {
-            if (!$this->isInStock($wishlistProduct)) {
-                continue;
+            if ($this->productCanBeProcessed($wishlistProduct)) {
+                $this->addProductToWishlist($wishlistProduct);
             }
-
-            $this->addProductToWishlist($wishlistProduct);
         }
     }
 
+    private function productCanBeProcessed(WishlistItem $wishlistProduct): bool
```

Comment:
Could you extract this into separate checker and spec it?

Reply:
Done.